SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/kyc/view/IdentityVerificationViewModel.kt

```diff
@@ -0,0 +1,271 @@
+            }
+        }
+        return verification
+    }
+
+    private fun checkCustomerStatus(state: CustomerBankModel.State) {
+
+        when (state) {
+
+            CustomerBankModel.State.storing -> {
+
+                if (customerJob == null) {
+                    customerJob = Pooling { getCustomerStatus() }
+                }
+            }
+
+            CustomerBankModel.State.verified -> {
+
+                customerJob?.stop()
+                UIState?.value = KYCView.KYCViewState.VERIFIED
+            }
+
+            CustomerBankModel.State.unverified -> {
+
+                customerJob?.stop()
+                getIdentityVerificationStatus()
+            }
+
+            CustomerBankModel.State.rejected -> {
+
+                customerJob?.stop()
+                UIState?.value = KYCView.KYCViewState.ERROR
+            }
+        }
+    }
+
+    private fun checkIdentityRecordStatus(record: IdentityVerificationBankModel?) {
+
+        when(record?.state) {
+
+            IdentityVerificationBankModel.State.storing -> {
+
+                if (identityJob == null) {
+                    identityJob = Pooling { getIdentityVerificationStatus(record = record) }
+                }
+            }
+
+            IdentityVerificationBankModel.State.waiting -> {
+
+                if (record.personaState != IdentityVerificationBankModel.PersonaState.completed) {
+
+                    identityJob?.stop()
+                    identityJob = null
+                    checkIdentityPersonaStatus(record)
+                } else {
+                    identityJob = Pooling { getIdentityVerificationStatus(record = record) }
+                }
+            }
+
+            IdentityVerificationBankModel.State.expired -> {
+
+                identityJob?.stop()
+                identityJob = null
+                getIdentityVerificationStatus(null)
+            }
+
+            IdentityVerificationBankModel.State.completed -> {
+
+                identityJob?.stop()
+                identityJob = null
+                UIState?.value = KYCView.KYCViewState.VERIFIED
+            }
+        }
+    }
+
+    private fun checkIdentityPersonaStatus(record: IdentityVerificationBankModel?) {
+
+        when(record?.personaState) {
+
+            IdentityVerificationBankModel.PersonaState.waiting -> {
+
+                this.latestIdentityVerification = record
+                UIState?.value = KYCView.KYCViewState.REQUIRED
+            }
+
+            IdentityVerificationBankModel.PersonaState.pending -> {
+
+                this.latestIdentityVerification = record
+                UIState?.value = KYCView.KYCViewState.REQUIRED
+            }
+
+            IdentityVerificationBankModel.PersonaState.reviewing -> {
+
+                UIState?.value = KYCView.KYCViewState.REVIEWING
+            }
+
+            IdentityVerificationBankModel.PersonaState.unknown -> {
+
+                UIState?.value = KYCView.KYCViewState.ERROR
+            }
```

Comment:
There are additional possible Persona states, listed below from the OpenAPI schema:
```
- waiting
- pending
- reviewing
- processing
- expired
- completed
- unknown
```

Reply:
The `expired` state i checked whe i get the IdentityVerification Status

The state for `porcessing` and `completed` is checked in this method `checkIdentityRecordStatus`